draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2937,263 @@ processing the server's Certificate, CertificateVerify, and
 Finished. If there is a failure in processing those messages, the
 client MUST discard the ServerConfiguration.
 
+### Authentication Messages
+
+As discussed in {{handshake-protocol-overview}}, TLS uses a common
+set of messages for authentication, key confirmation, and handshake
+integrity: Certificate, CertificateVerify, and Finished. These
+messages are always sent as the last messages in their handshake
+flight. The Certificate and CertificateVerify messages are only
+sent under certain circumstances, as defined below. The Finished
+message is always sent as part of the Authentication block.
+
+The computations for the Authentication messages all uniformly
+take the following inputs:
+
+- The certificate and signing key to be used.
+- A Handshake Context based on the handshake hash (see {{the-handshake-hash}}).
+- A base key to be used to compute a MAC key.
+
+Based on these inputs, the messages then contain:
+
+Certificate
+: The certificate to be used for authentication and any
+supporting certificates in the chain.
+
+CertificateVerify
+: A signature over Handshake Context + Certificate.
+
+Finished
+: A MAC over Handshake Context + Certificate + CertificateVerify using
+  a MAC key derived from the base key.
+{:br}
+
+Because the CertificateVerify signs the Handshake Context +
+Certificate and the Finished MACs the Handshake Context + Certificate
++ CertificateVerify, this is mostly equivalent keeping a running hash
+of the handshake messages (exactly so in the pure 1-RTT cases). Note,
+however, the subsequent post-handshake authentications do not include
+each other, just the messages through the end of the main handshake.
+
+The following table defines the Handshake Context and MAC Key
+for each scenario:
+
+~~~~
+Mode             Handshake Context                        Base Key
+----             -----------------                        --------
+0-RTT            ClientHello + ServerConfiguration        xSS
+                             + Server Certificate
+                             + CertificateRequest
```

Comment:
It would be easier on clients if this were `H(ClientHello) || H(CertificateRequest || ServerConfiguration) || H(ServerCertificate)` or maybe `H(ClientHello) || H(CertificateRequest) || H(ServerConfiguration) || H(ServerCertificate)` I think.  There are things from each of those messages that they need to save, but a lot of that can be reduced if you can just store a hash.